Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1112,8 +1188,9 @@ bool LineBuilder::tryPlacingFloatBox(const InlineItem& floatItem, LineBoxConstra
     boxGeometry.setLogicalTopLeft(staticPosition);
     // Float it.
     ASSERT(m_rootHorizontalConstraints);
-    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
     auto floatingPosition = floatingContext.positionForFloat(floatBox, *m_rootHorizontalConstraints);
+    if (!containingBlockMarginTrim.isEmpty())
+        trimMarginsForPositionedFloat(floatingPosition);
```

Comment:
You've certainly thought about it more than I did, so I am curious as to when would a float end up in flush with the root's content box and not yet trimmed (see tryTrimmingFloatInlineMarginsAccordingToConstraints in shouldBePlaced -though I think you missed the lineIsConsideredEmpty case in there.)